From b0cb897fe9fc4d29b07eb1356d61f641cbe594a1 Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@linux.intel.com>
Date: Fri, 2 Dec 2011 13:55:36 +0200
Subject: [PATCH] wifi: Disable network in disconnected state

Disable WiFi network when it ends up in disconnected state
in order to prevent wpa_supplicant looping forever retrying.

Fixes BMC#23973
---
 plugins/wifi.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/plugins/wifi.c b/plugins/wifi.c
index 745c2cb..538230a 100644
--- a/plugins/wifi.c
+++ b/plugins/wifi.c
@@ -912,12 +912,6 @@ static connman_bool_t handle_4way_handshake_failure(GSupplicantInterface *interf
 	if (wifi->retries < MAXIMUM_RETRIES)
 		return TRUE;
 
-	/* We disable the selected network, if not then
-	 * wpa_supplicant will loop retrying */
-	if (g_supplicant_interface_enable_selected_network(interface,
-								FALSE) != 0)
-		DBG("Could not disables selected network");
-
 	connman_network_set_error(network, CONNMAN_NETWORK_ERROR_INVALID_KEY);
 
 	return FALSE;
@@ -987,6 +981,12 @@ static void interface_state(GSupplicantInterface *interface)
 						network, wifi) == TRUE)
 			break;
 
+		/* We disable the selected network, if not then
+		 * wpa_supplicant will loop retrying */
+		if (g_supplicant_interface_enable_selected_network(interface,
+						FALSE) != 0)
+			DBG("Could not disables selected network");
+
 		connman_network_set_associating(network, FALSE);
 		connman_network_set_connected(network, FALSE);
 		break;
-- 
1.7.9

